src/components/user/user.service.ts

```diff
@@ -364,6 +372,48 @@ export class UserService {
       nodevar: 'user',
     });
 
+    // Update email
+    if (input.email) {
+      // Remove old emails and relations
+      const uniqueProperties: UniqueProperties<User> = {
+        email: ['Property', 'EmailAddress'],
+      };
+      await this.db
+        .query()
+        .match([node('node', ['User', 'BaseNode'], { id: user.id })])
+        .call(setPropLabelsAndValuesDeleted, uniqueProperties)
+        .return('*')
+        .run();
+
+      try {
+        const createdAt = DateTime.local();
+        await this.db
+          .query()
+          .match([node('user', ['User', 'BaseNode'], { id: user.id })])
+          .create([
+            node('user'),
+            relation('out', '', 'email', {
+              active: true,
+              createdAt,
+            }),
+            node('email', 'EmailAddress:Property', {
+              value: input.email,
+              createdAt,
+            }),
+          ])
+          .run();
+      } catch (e) {
+        if (e instanceof UniquenessError && e.label === 'EmailAddress') {
+          throw new DuplicateException(
+            'person.email',
+            'Email address is already in use',
+            e
+          );
+        }
+        throw new ServerException('Failed to create user', e);
+      }
+    }
```

Comment:
Since there's the potential for the user's email to be deleted only to determine that the new email address is already in use means we are leaving the db in a broken state.

Thankfully, I think we can just add the `@Transactional()` decorator to this update() method which should rollback the entire transaction when we throw an error.